src/Controls/src/SourceGen/GeneratorHelpers.cs

```diff
@@ -398,4 +398,31 @@ static void SimplifyOnPlatform(XmlNode node, string? targetFramework, XmlNamespa
 		}
 		return null;
 	}
+
+	/// <summary>
+	/// Formats a numeric value as a culture-independent C# literal for source generation.
+	/// Uses SymbolDisplay.FormatPrimitive to ensure proper handling of special values like NaN and Infinity.
+	/// </summary>
+	public static string FormatInvariant(double value)
+	{
+		return SymbolDisplay.FormatPrimitive(value, false, false);
+	}
+
+	/// <summary>
+	/// Formats a numeric value as a culture-independent C# literal for source generation.
+	/// Uses SymbolDisplay.FormatPrimitive to ensure proper handling of special values like NaN and Infinity.
+	/// </summary>
+	public static string FormatInvariant(float value)
+	{
+		return SymbolDisplay.FormatPrimitive(value, false, false);
+	}
+
+	/// <summary>
+	/// Formats a numeric value as a culture-independent C# literal with quote strings enabled.
+	/// Uses SymbolDisplay.FormatPrimitive to ensure proper handling of special values like NaN and Infinity.
+	/// </summary>
+	public static string FormatInvariantWithQuotes(double value)
+	{
+		return SymbolDisplay.FormatPrimitive(value, true, false);
+	}
```

Comment:
multiple things here....
1/ I wouldn't add FormatInvariant helpers, I would instead name the boolean parameters in the invocation
2/ If you insist on providing helpers, they should accept an object as value (like FormatPriomitive), and the withQuotes should be an optional parameter

Reply:
Kept the helper method. We're probably never going to use the third parameter and having that everywhere, including the parameter name seems a bit excessive. Addressed the rest of the feedback though, thanks!